### PR TITLE
Add if condition to workflow job and implement consistent naming

### LINKED ISSUE #846: Implement if condition in workflow jobs
# 💡 Summary #

A specific "if condition" can be added to workflow jobs to prevent duplicate execution of workflows.  The condition was successfully tested in https://github.com/cisagov/ScubaGear/issues/805, so now it's time to implement this condition in the ScubaGear workflows.

This is a follow-on to: https://github.com/cisagov/ScubaGear/issues/805
This is part of an epic: https://github.com/cisagov/ScubaGear/issues/819

## Motivation and context ##

This would be useful because unnecessary and redundant workflow executions are unnecessary and redundant.

## Implementation notes ##

- Look for workflows that have push and/or pull request.
- Make sure they have both.
- Add the if condition to all jobs
- Test the workflows.

## Acceptance criteria ##

How do we know when this work is done?

- [ ] All workflows with the condition still run when there's a push.

### LINKED ISSUE #859: Provide meaningful and consistent names for all workflows
# 💡 Summary #

Rename the workflows such that their names are consistent (e.g., always start with a verb) and are indicative of what they actually do (e.g., unit test workflows should say unit test).

This is part of an epic:  https://github.com/cisagov/ScubaGear/issues/819

## Motivation and context ##

This would be useful because inconsistent and unmeaningful names are inconsistent and unmeaningful.

## Implementation notes ##

- rename each workflow
- verify that they still run

## Acceptance criteria ##

- [ ] Each workflow has a meaningful and consistent name

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

Adds an "if condition" to all workflow jobs to prevent duplicate workflow runs.
Cleans up the workflows by enforcing some consistent naming and file order conditions.

## 💭 Motivation and context ##

Workflows that trigger on push and pull requests ran twice.  A previous issue (https://github.com/cisagov/ScubaGear/issues/805) found a way to prevent this using an "if condition", so this was implemented.

Multiple developers have used various naming conventions and workflow file organization conventions when writing workflows, making them a bit more difficult to read and understand.  This simply cleaned the workflows up by enforcing a consistent style.

Closes  #846
Closes  #859 

## 🧪 Testing ##

This was tested by running all the workflows again, and they all passed.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

**Review by crutchfield — APPROVED** [review]
Looks good

**Review by dagarwal-mitre — APPROVED** [review]

File: .github/workflows/build_sign_release.yaml [hunk 1]
```diff
@@ -15,17 +19,18 @@ on:
         type: boolean
         default: true
 
-name: Build and Sign Release
-
 jobs:
   build-and-deploy:
+    name: Build and Deploy
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/build_sign_release.yaml:24]: Done.

File: .github/workflows/smoke_test.yaml
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/smoke_test.yaml:1]: I contend that anything, if used properly, can be a verb.    :-)

In this case, "test" is the verb and "smoke" is the type of test.  Same as the workflows "unit test opt" and "unit test powershell".  

Even so, I have added "run" as the verb.

**Comment by james-garriss** [comment]
Smoke test is failing because it's missing an HTML file.  This does not appear to be related to my code AFAICT.

File: .github/workflows/build_sign_release.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/build_sign_release.yaml]: Done

**Comment by james-garriss** [comment]
@nanda-katikaneni , I think this one is ready.

**Comment by james-garriss** [comment]
@nanda-katikaneni , although publish package is failing, that is unrelated to my code or this PR.  Crutch has opened a new issue (#973) to deal with that.  So I think this is still ready for review and merge.

**Pull request merged**